slugid.js

```diff
@@ -42,13 +46,16 @@ exports.decode = function(slug) {
   return uuid.unparse(new Buffer(base64, 'base64'));
 };
 
-/** Generate a v4 (random) uuid and encode it to a slug */
+/** Returns a randomly generated uuid v4 complaint slug guaranteed not to have
+ * a leading `-` character */
 exports.v4 = function() {
-  var bytes   = uuid.v4(null, new Buffer(16));
-  var base64  = bytes.toString('base64');
+  do {
+    var bytes   = uuid.v4(null, new Buffer(16));
+    var base64  = bytes.toString('base64');
+  } while (/^\+/.test(base64));  // disallow leading '-' ('+' => '-' below)
```

Comment:
> Using crypto.randomBytes(1) and accounting for potential errors (it can throw an error if entropy
> sources are drained)

No it, cannot in practice... see: http://stackoverflow.com/questions/18130254/randombytes-vs-pseudorandombytes

In fact the `uuid` module uses `crypto.randombytes()`... Note, I'm also okay hardcoding it...